tracim/tracim/lib/email_fetcher.py

```diff
@@ -0,0 +1,323 @@
+
+    def get_body(self) -> typing.Optional[str]:
+        body_part = self._get_mime_body_message()
+        body = None
+        if body_part:
+            charset = body_part.get_content_charset('iso-8859-1')
+            content_type = body_part.get_content_type()
+            if content_type == CONTENT_TYPE_TEXT_PLAIN:
+                txt_body = body_part.get_payload(decode=True).decode(
+                    charset)
+                body = DecodedMail._parse_txt_body(txt_body)
+
+            elif content_type == CONTENT_TYPE_TEXT_HTML:
+                html_body = body_part.get_payload(decode=True).decode(
+                    charset)
+                body = DecodedMail._parse_html_body(html_body)
+
+        return body
+
+    @classmethod
+    def _parse_txt_body(cls, txt_body: str):
+        txt_body = EmailReplyParser.parse_reply(txt_body)
+        html_body = markdown.markdown(txt_body)
+        body = DecodedMail._parse_html_body(html_body)
+        return body
+
+    @classmethod
+    def _parse_html_body(cls, html_body: str):
+        soup = BeautifulSoup(html_body,'html.parser')
+        config = BEAUTIFULSOUP_HTML_BODY_PARSE_CONFIG
+        for tag in soup.findAll():
+            if DecodedMail._tag_to_extract(tag):
+                tag.extract()
+            elif tag.name.lower() in config['tag_whitelist']:
+                attrs = dict(tag.attrs)
+                for attr in attrs:
+                    if attr not in config['attrs_whitelist']:
+                        del tag.attrs[attr]
+            else:
+                tag.unwrap()
+        return str(soup)
+
+    @classmethod
+    def _tag_to_extract(cls, tag) -> bool:
+        config = BEAUTIFULSOUP_HTML_BODY_PARSE_CONFIG
+        if tag.name.lower() in config['tag_blacklist']:
+            return True
+        if 'class' in tag.attrs:
+            for elem in config['class_blacklist']:
+                if elem in tag.attrs['class']:
+                    return True
+        if 'id' in tag.attrs:
+            for elem in config['id_blacklist']:
+                if elem in tag.attrs['id']:
+                    return True
+        return False
+
+    def _get_mime_body_message(self) -> typing.Optional[Message]:
+        # TODO - G.M - 2017-11-16 - Use stdlib msg.get_body feature for py3.6+
+        part = None
+        # Check for html
+        for part in self._message.walk():
+            content_type = part.get_content_type()
+            content_dispo = str(part.get('Content-Disposition'))
+            if content_type == CONTENT_TYPE_TEXT_HTML \
+                    and 'attachment' not in content_dispo:
+                return part
+        # check for plain text
+        for part in self._message.walk():
+            content_type = part.get_content_type()
+            content_dispo = str(part.get('Content-Disposition'))
+            if content_type == CONTENT_TYPE_TEXT_PLAIN and 'attachment' \
+                    not in content_dispo:
```

Comment:
Instead of

```
         if content_type == CONTENT_TYPE_TEXT_PLAIN and 'attachment' \
                 not in content_dispo:
```

It's more readable to split line on end of condition:

```
         if content_type == CONTENT_TYPE_TEXT_PLAIN \
                  and 'attachment' not in content_dispo:
```

What do you think?